vulnerabilities/memory-corruption-due-to-insufficient-allocation.md

```diff
@@ -0,0 +1,175 @@
+# Memory Corruption Due to Insufficient Memory Allocation
+
+Memory corruption occurs when a program writes outside the bounds of allocated memory, potentially overwriting other important data.
```

Comment:
I think it would be more accurate to say, "Memory corruption can occur..." since it's not the only cause

Reply:
fair enough, lemme see if we can generalize this vuln type as suggested